libs/wingsdk/src/util/util.ts

```diff
@@ -35,6 +70,32 @@ export class Util {
     return new Promise((resolve) => setTimeout(resolve, delay.seconds * 1000));
   }
 
+  /**
+   * Waits until predicate is true or return false
+   * @param predicate The function that will be evaluated
+   * @param props Timeout and interval values, default to one 1m timeout and 0.1sec interval
+   * @inflight
+   */
+  public static async wait(
+    predicate: IPredicateHandler,
+    props: WaitProps = {}
+  ): Promise<boolean> {
+    const timeout = props.timeout ?? Duration.fromMinutes(1);
+    const interval = props.interval ?? Duration.fromSeconds(0.1);
+    const f = predicate as any;
+    let elapsed = 0;
+    while (elapsed < timeout.seconds) {
+      if (await f()) {
```

Comment:
Should we surround the f invocation with try catch and treat exception as false?

Reply:
I could see it going either way. Maybe propagating the error is fine? Let's document it either way, e.g.:

`@throws if the given predicate throws`

Reply:
I think we should propagate the error. There's no "happy reason" for the predicate to throw.

Reply:
Not sure that's a super convincing example 🤷🏻‍♀️

I okay if we want to add an option:

```js
let p = () => { return b.get("key") == "value"; };
util.waitUntil(p, ignoreErrors: true);
```

This example also brings up an idea related to optionality: perhaps we should allow comparing `T?` and `T`? Seems like a safe thing to allow...

Reply:
> ignoreErrors

Not doing this for now, let see if it surface

Reply:
Right now, it throws  (added the `@throws Will throw if the given predicate throws.`)